sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncDatabase.java

```diff
@@ -815,7 +815,16 @@ public CosmosPagedFlux<CosmosContainerProperties> queryContainers(SqlQuerySpec q
      * @return Cosmos Container
      */
     public CosmosAsyncContainer getContainer(String id) {
-        return new CosmosAsyncContainer(id, this);
+        CosmosAsyncContainer asyncContainer = this
+            .client
+            .getContainerCreationInterceptor()
+            .apply(new CosmosAsyncContainer(id, this));
```

Comment:
if a user is doing an I/O call on this containerCreationInterceptor, this overall `getContainer()` call becomes a network call, which we might mistakenly use internally for a non-network call. 
Also, based on what errors end user's function call might throw, I am curious, how shall we accomodate those? to me this looks like similar error handling case of how we exposed custom diagnostics handler.

Reply:
I don't think that really is an issue - GetContainer is a sync API today - so, when the app calls it we can't accomodate async interception - so, this is simply not a supported scenario. Same with exceptions - in this case ti would just bubble up tp app - which is fine IMO - unlike diagnostics handler we don't expect this to be used widely. I also don't think we really can interfere with exception creation here - there is no way to finish the API call without exception if the interceptor is regsitered and throws.